src/Rings/mpoly-ideals.jl

```diff
@@ -2249,3 +2277,11 @@ function flag_pluecker_ideal(ring::MPolyRing{<: FieldElem}, dimensions::Vector{I
                    isReduced=true,
                    isGB=true))
 end
+
+# Since most ideals implement `==`, they have to implement the hash function.
+# See issue #4143 for problems entailed. Interestingly, this does not yet fix 
+# the failure of unique! on lists of ideals.
+function hash(I::Ideal, c::UInt)
+  return UInt(0)
+end
```

Comment:
```suggestion
function hash(I::Ideal, h::UInt)
  return h
end
```
returning the second value is save and a small optimisation.

Reply:
Hashing the concrete type and maybe even the parent ring should also be fast and help avoid a few comparisons.